src/Microsoft.OpenApi.Readers/V2/OpenApiV2Deserializer.cs

```diff
@@ -29,10 +29,11 @@ internal static partial class OpenApiV2Deserializer
                 return;
             }
 
-            foreach (var propertyNode in mapNode)
+            var allFields = fixedFieldMap.Keys.Union(mapNode.Select(x => x.Name));
+            foreach (var propertyNode in allFields)
             {
-                propertyNode.ParseField(domainObject, fixedFieldMap, patternFieldMap);
-                requiredFields?.Remove(propertyNode.Name);
+                mapNode[propertyNode]?.ParseField(domainObject, fixedFieldMap, patternFieldMap);
```

Comment:
try get instead with a condition?

Reply:
MapNode implements an IEnumerable <T> hence doesn't expose a TryGetValue() method

Reply:
ah, I can see the implementation already does a TryGetValue https://github.com/microsoft/OpenAPI.NET/blob/5b080619be23f2b9e2e3995f53393f2263e5a4e3/src/Microsoft.OpenApi.Readers/ParseNodes/MapNode.cs#L46